sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/TagsExtension.cs

```diff
@@ -49,33 +52,69 @@ internal static class TagsExtension
             [SemanticConventions.AttributeMessagingUrl] = PartBType.Messaging
         };
 
-        internal static Dictionary<string, string> ToAzureMonitorTags(this IEnumerable<KeyValuePair<string, string>> tags, out PartBType activityType)
+        internal static PartBType ToAzureMonitorTags(this IEnumerable<KeyValuePair<string, object>> tagObjects, out Dictionary<string, string> partBTags, out Dictionary<string, string> partCTags)
         {
-            Dictionary<string, string> partBTags = new Dictionary<string, string>();
-            activityType = PartBType.Unknown;
+            if (tagObjects == null)
+            {
+                partBTags = null;
+                partCTags = null;
+                return PartBType.Unknown;
+            }
+
+            PartBType tempActivityType;
+            PartBType activityType = PartBType.Unknown;
+            partBTags = new Dictionary<string, string>();
+            partCTags = new Dictionary<string, string>();
 
-            foreach (var entry in tags)
+            foreach (var entry in tagObjects)
             {
-                // TODO: May need to store unknown to write to properties as Part C
-                if ((activityType == PartBType.Unknown || activityType == PartBType.Net) && !Part_B_Mapping.TryGetValue(entry.Key, out activityType))
+                if (entry.Value is Array array)
                 {
-                    if (activityType == PartBType.Net)
+                    StringBuilder sw = new StringBuilder();
+                    foreach (var item in array)
                     {
-                        partBTags.Add(entry.Key, entry.Value);
-                        activityType = PartBType.Unknown;
+                        if (item != null)
+                        {
+                            sw.Append(item);
```

Comment:
Curious, do we want this to be a comma separated string or a JSON string?

Reply:
Nice question, JSON array will look better than comma separation. I will get this changed.